File: app/src/main/java/com/theappsolutions/boilerplate/TasBoilerplateSettings.java
package com.theappsolutions.boilerplate;

import android.Manifest;

public class TasBoilerplateSettings {

    public static final String API = "https://dev.test.org/";
    public static final String API_VERSION = "v1";

    public static final int MIN_PASSWORD_LENGTH = 6;
    public static final boolean PASSWORD_SHOULD_HAVE_NUMBERS_AND_LETTERS = true;
    public static final boolean PASSWORD_SHOULD_HAVE_UPPERCASE_LETTERS = true;

    public static long REALM_LOCAL_DB_VERSION = 1;
    public static final String REALM_DB_NAME = "tas_boilerplate.realm";


    public static final String[] NECESSARY_PERMISSIONS = {
            android.Manifest.permission.INTERNET,
            android.Manifest.permission.ACCESS_NETWORK_STATE,
            android.Manifest.permission.ACCESS_FINE_LOCATION,
            android.Manifest.permission.WRITE_EXTERNAL_STORAGE,
            Manifest.permission.READ_EXTERNAL_STORAGE
    };

    public static class TestSettings {
        public static final int RESPONSE_ELEMENT_COUNT = 3;
    }
}

File: app/src/main/java/com/theappsolutions/boilerplate/TasBoilerplateSettings.kt
package com.theappsolutions.boilerplate

import android.Manifest

/**
 * @author Severyn Parkhomenko s.parkhomenko@theappsolutions.com
 * @copyright (c) 2018 TheAppSolutions. (https://theappsolutions.com)
 */
object TasBoilerplateSettings {

    val API = "https://dev.test.org/"
    val API_VERSION = "v1"

    val MIN_PASSWORD_LENGTH = 6
    val PASSWORD_SHOULD_HAVE_NUMBERS_AND_LETTERS = true
    val PASSWORD_SHOULD_HAVE_UPPERCASE_LETTERS = true

    var REALM_LOCAL_DB_VERSION: Long = 1
    val REALM_DB_NAME = "tas_boilerplate.realm"

    val SPLASH_SCREEN_DELAY_SEC = 2

    val NECESSARY_PERMISSIONS = arrayOf(android.Manifest.permission.INTERNET,
            android.Manifest.permission.ACCESS_NETWORK_STATE,
            android.Manifest.permission.ACCESS_FINE_LOCATION,
            android.Manifest.permission.WRITE_EXTERNAL_STORAGE,
            Manifest.permission.READ_EXTERNAL_STORAGE)

    object TestSettings {
        val RESPONSE_ELEMENT_COUNT = 3
    }
}
